lib/Reconciler.lua

```diff
@@ -93,8 +105,10 @@ function Reconciler.unmount(instanceHandle)
 
 		-- Kill refs before we make changes, since any mutations past this point
 		-- aren't relevant to components.
-		if element.props[Core.Ref] then
-			element.props[Core.Ref](nil)
+		local ref = element.props[Core.Ref]
+
+		if ref then
+			applyRef(ref, nil)
```

Comment:
Let's move this condition into `applyRef` itself, which would make all of these invocations a bit cleaner!

Reply:
That'd help clean some stuff up, yeah!